fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -1202,7 +1202,9 @@ ACTOR Future<Void> BgDDMountainChopper( DDQueueData* self, int teamCollectionInd
 				state Optional<Reference<IDataDistributionTeam>> randomTeam = wait(brokenPromiseToNever(
 				    self->teamCollections[teamCollectionIndex].getTeam.getReply(GetTeamRequest(true, false, true))));
 				if (randomTeam.present()) {
-					if (randomTeam.get()->getMinFreeSpaceRatio() > SERVER_KNOBS->FREE_SPACE_RATIO_DD_CUTOFF) {
+					// Destination team must be healthy and have healthyFreeSpace, otherwise, BestTeamStuck may occur
+					if (randomTeam.get()->getMinFreeSpaceRatio() > SERVER_KNOBS->FREE_SPACE_RATIO_DD_CUTOFF &&
+					    randomTeam.get()->hasHealthyFreeSpace()) {
```

Comment:
Interesting, my interpretation is that this could only be a problem when `MIN_FREE_SPACE` is more than 20% of capacity (or you've done some weird knob setting), and at 100MB that would require a capacity of less than 500MB. Is this something that has only showed up in simulation?

Reply:
This PR was because one simulation test found itself stuck at `getTeam()`.
In [`getTeam`](https://github.com/apple/foundationdb/blob/master/fdbserver/DataDistribution.actor.cpp#L786), it checks `hasHealthyFreeSpace()`.